microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -40,96 +55,133 @@ As explained in [Fix memory problems](index.md), memory issues affect page perfo
 
    ![Analyze Detached Elements in the Detached Elements tool](./dom-leaks-images/analyze-detached-elements.png)
 
-1. From the **Detached Elements** tool, select the **Id** field of a detached element.
+1. In the **Detached Elements** tool, in the **Id** column, click **@21299**, which is the ID of one of the `<div class="message">` elements.
 
-   The **Memory** tool automatically selects the object in the heap that is referencing the detached element. We call these objects **Retainers**:
+   The **Memory** tool automatically selects the object in the heap that is referencing the detached element.  These objects are called _retainers_, and are listed in the **Retainers** tab:
 
    ![Referencing a heap snapshot from the Detached Elements tool](./dom-leaks-images/heap-snapshot.png)
 
-1. In the **Memory** tool, select the link **room.js:13**.
-
-   The **Sources** tool opens in the **Activity Bar** and shows line 13 of the file **room.js**.
+1. In the **Memory** tool, in the **Retainers** tab, in the **[2] in Array @58525** retainer entry, in the **unmounted in Room @57707** retainer sub-entry, click the link **room.js:13**.
 
-1. In the `hide()` function of **room.js**, the JavaScript code of the demo application adds each message in the room to an `unmounted` array. The `unmounted` array is the object that is referencing the detached element:
+   The **Sources** tool opens in the **Activity Bar** and shows line 13 of the file **room.js**:
 
-   ![Identifying the JavaScript that is retaining the detached element](./dom-leaks-images/javascript-retainers.png)
+   ![Line 13 in room.js](./dom-leaks-images/room-js-line-13.png)<!-- todo: redo png to show line 13 -->
 
-You have now identified the retainer that is preventing the detached element from being garbage-collected by the browser!
+   Scroll down to line 49:<!-- todo: why? -->
 
+   ![Identifying the JavaScript that is retaining the detached element](./dom-leaks-images/javascript-retainers.png)
 
-<!-- ====================================================================== -->
-## Identify the DOM node causing others to be retained
-
-Because the DOM is a fully connected graph, when one DOM node is retained in memory by JavaScript it can cause other DOM nodes to be retained with it.
+   In the `hide()` function of **room.js**, the JavaScript code of the demo application adds each message in the room to an `unmounted` array. The `unmounted` array is the object that is referencing the detached element.
 
-To identify the culprit node in a detached tree that is causing the entire tree to be retained:
+   In the JavaScript code, you have now identified the retainer that is preventing the detached element from being garbage-collected by the browser.  Next, identify the DOM node causing others to be retained, as follows.
 
 1. Click the **Detach Elements** (![The Detach Elements icon](./dom-leaks-images/detach-elements-icon.png)) icon to destroy the parent-child links inside the detached tree.
```

Comment:
15. Click the Detach Elements (The Detach Elements icon) icon to destroy the parent-child links inside the detached tree.   [_why?_]
16. Click the Collect garbage (The 'Collect garbage' icon) icon.   [_why?_]
Parent-child links are removed inside the detached tree:   [_why? so what?  what's the value or point of this?_]
@captainbrosset 

why?  to accomplish what?  what's the strategy here?

steps at end of section:
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks?branch=pr-en-us-3282&tabs=detached-elements-tool#open-the-demo-and-tool-get-detached-elements-and-analyze-a-detached-element